### PR TITLE
ci: Update csi_upgrade_version

### PR DESCRIPTION
**Pull request by karthik-us**
Point the csi_upgrade_version from v3.7.2 to the
latest release v3.9.0.

Updates: #3911

### CODE REVIEW

**Review by Madhu-1 — COMMENTED** [review]
Please note, as a follow-up, the second TODO need to be addressed. Can you create a draft PR for the second 2nd TODO so that we can block the PR merging quickly after merging this one?

**Review by Madhu-1 — APPROVED** [review]

**Review by Rakshith-R — APPROVED** [review]

**Pull request merged**
